pkg/eth/tx_listener.go

```diff
@@ -88,8 +114,22 @@ func (t *TxListener) PollForTxCompletion(txID string, updates chan<- string) {
 
 }
 
-func (t *TxListener) checkTx(hash common.Hash) (bool, error) {
+func (t *TxListener) waitForTimeout(w *waiter) {
+	timer := time.NewTimer(t.timeout)
+	select {
+	case <-timer.C:
+		w.Mutex.Lock()
+		w.DoneWaiting = true
+		w.Mutex.Unlock()
+		return
```

Comment:
Minor, but you don't need the `return` in both these cases, it should just exit.